scripts/copy_from_upstream/patches/pqclean-kyber-arm-macos-gcc-fix.patch

```diff
@@ -0,0 +1,318 @@
+@@ -4,7 +4,7 @@
+ .align 2
+ .global PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_top
+ .global _PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_top
+-#ifndef __clang__
++#if !defined(__clang__) && !defined(old_gas_syntax)
+ .type PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_top, %function
+ #endif
+ PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_top:
+@@ -173,7 +173,7 @@ _PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_top:
+ .align 2
+ .global PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_bot
+ .global _PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_bot
+-#ifndef __clang__
++#if !defined(__clang__) && !defined(old_gas_syntax)
+ .type PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_bot, %function
+ #endif
+ PQCLEAN_KYBER1024_AARCH64_asm_ntt_SIMD_bot:
+diff --git a/crypto_kem/kyber1024/aarch64/__asm_base_mul.S b/crypto_kem/kyber1024/aarch64/__asm_base_mul.S
+index 94ba36b..3eed305 100644
+--- a/crypto_kem/kyber1024/aarch64/__asm_base_mul.S
++++ b/crypto_kem/kyber1024/aarch64/__asm_base_mul.S
+@@ -6,7 +6,7 @@
+ .align 2
+ .global PQCLEAN_KYBER1024_AARCH64_asm_point_mul_extended
+ .global _PQCLEAN_KYBER1024_AARCH64_asm_point_mul_extended
+-#ifndef __clang__
++#if !defined(__clang__) && !defined(old_gas_syntax)
+   .type PQCLEAN_KYBER1024_AARCH64_asm_point_mul_extended, %function
+ #endif
+ PQCLEAN_KYBER1024_AARCH64_asm_point_mul_extended:
+@@ -73,7 +73,7 @@ _PQCLEAN_KYBER1024_AARCH64_asm_point_mul_extended:
+ .align 2
+ .global PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul
+ .global _PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul
+-#ifndef __clang__
++#if !defined(__clang__) && !defined(old_gas_syntax)
+ .type PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul, %function
+ #endif
+ PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul:
+@@ -228,7 +228,7 @@ _PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul:
+ .align 2
+ .global PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul_montgomery
+ .global _PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul_montgomery
+-#ifndef __clang__
++#if !defined(__clang__) && !defined(old_gas_syntax)
+ .type PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul_montgomery, %function
+ #endif
+ PQCLEAN_KYBER1024_AARCH64_asm_asymmetric_mul_montgomery:
+diff --git a/crypto_kem/kyber1024/aarch64/__asm_iNTT.S b/crypto_kem/kyber1024/aarch64/__asm_iNTT.S
+index 5701058..f7e83ab 100644
+--- a/crypto_kem/kyber1024/aarch64/__asm_iNTT.S
++++ b/crypto_kem/kyber1024/aarch64/__asm_iNTT.S
+@@ -4,7 +4,7 @@
+ .align 2
+ .global PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_bot
+ .global _PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_bot
+-#ifndef __clang__
++#if !defined(__clang__) && !defined(old_gas_syntax)
+ .type PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_bot, %function
+ #endif
+ PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_bot:
+@@ -90,7 +90,7 @@ _PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_bot:
+ .align 2
+ .global PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_top
+ .global _PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_top
+-#ifndef __clang__
++#if !defined(__clang__) && !defined(old_gas_syntax)
+ .type PQCLEAN_KYBER1024_AARCH64_asm_intt_SIMD_top, %function
```

Comment:
This doesn't look like a SABER-related change. Where does this come from?

Reply:
The patch was previously called `pqclean-saber-kyber-arm-macos-gcc-fix.patch`. I removed all the saber related changes, created a new patch called `pqclean-kyber-arm-macos-gcc-fix.patch`, and removed the old patch.

Reply:
Ahh -- I was confused by the deltas -- but it's clearly a new file. Thanks for the explanation